backend/xivdpscalc/pps/sch.py

```diff
@@ -12,6 +17,71 @@ class SchPps(HealerPps):
     bio_potency = 70
     ed_potency = 100
 
+    def get_total_potency_variable_time(self, sim_length: ElapsedTime, character_stats: Character, rotation: SchRotation, caster_tax: float = 0.1) -> float:
+        """
+        Get the total potency in sim_length seconds
+        rotation is a SchRotation object
+        todo: support dot damage calculation
+        todo: how long is animation lock actually?  Hard coded here for 0.75, have no idea if that's true
```

Comment:
i am so fucking sorry you asked this question but uh you're not gonna like the answer:

animation lock for actions varies (lol old drg jump lol). most actions have a base animation lock of 600ms.
HOWEVER. animation lock is actually ping dependent. the real animation lock is (600ms + ping + server delay), and server delay is about 40ms. (source: https://twitter.com/perchbird_/status/1282734093600522241 has the right idea but is missing the server delay and has the wrong lock time)

so i think that for animation lock, you should have this function take an additional optional ping parameter (default = 75) and compute animation lock based off of it.

before you ask, no i do not know the base animation lock values for each action, but i do know where to find them. take 600 base for now and make a todo to add in real values later.